core/src/test/java/com/google/zxing/common/AbstractBlackBoxTestCase.java

```diff
@@ -92,6 +93,10 @@ protected final void addTest(int mustPassCount, int tryHarderCount, float rotati
     addTest(mustPassCount, tryHarderCount, 0, 0, rotation);
   }
 
+  protected void addHint(DecodeHintType hint) {
```

Comment:
I'm worried it modifies a hints object reused by all tests, so one test calling this affects the next, but, looks like it's only used where it's global to all tests in a suite object.

Reply:
I chose this method because it looked liked the way with the minimal impact to pass a hint to the decoder.

The hint object is created deep down in the decode method and fitted with several test related values. What I'd like to have used is a parameter to pass an arbitrary hint to the addTest method. I think that would be helpful for other tests, too. I just didn't felt I understand the logic of the test class sufficently enough to introduce that myself.